src/main/cpp/commands/arm/IncrementArmPresetPositionCommand.cpp

```diff
@@ -27,7 +27,12 @@ void IncrementArmPresetPositionCommand::Execute() {
     VOKC_CALL(arm_->IncrementRotation(increment_));
 }
 
-bool IncrementArmPresetPositionCommand::IsFinished() {
-    // Always end this command.
-    return true;
-}
\ No newline at end of file
+
+    
+    bool IncrementArmPresetPositionCommand::IsFinished() {
+    bool at = false;
+
+    OKC_CALL(arm_->AtExtendSetpoint(&at));
+
+    return at;
```

Comment:
same comment as before - can't we make a "latching" state in the state machine and then activate a "retract" command when the button is released to retract everything. This way we can manage all these cases in the subsystem rather than relying on commands